### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by shruthipatil11**
Added Landing Page. 

Note: Ignore the images and dynamic data loading for Trending and collections in my landing page as they will be added later.

1. Pages Modified: 
              a.  https://github.com/shruthipatil11/oasis-mart/tree/home-page/src/pages/home-page
              b.  https://github.com/shruthipatil11/oasis-mart/tree/home-page/src/components/context
              c. https://github.com/shruthipatil11/oasis-mart/tree/home-page/src/components/category
              d. https://github.com/shruthipatil11/oasis-mart/tree/home-page/src/components/reducers
 
Please review.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,41 +1,17 @@
 import "./App.css";
 import logo from "./logo.png";
+import {Routes,Route,Link} from 'react-router-dom';
```
Comment (Debarshi95) [inline src/App.js:3]: Add whitespaces between comma, and braces.

File: src/assets/images/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import bags from '../images/bags.jpg';
+import footWear from '../images/foot wear.jpg';
+import tshirt from '../images/tshirt.jpg';
+import uiLogo from '../images/ui-logo.jpg';
+import watch from '../images/watch.jpg';
+import westernWear from '../images/western-wear.jpg';
+
+export {bags,footWear,tshirt,uiLogo,watch,westernWear};
```
Comment (Debarshi95) [inline src/assets/images/index.js]: Same here. Add whitespace between commas and the brackets. Use a formatter like prettier and setup your editor to format on save.

File: src/pages/home-page/HomePage.jsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+import "./home-page.css";
+import {Category,Header,useDataFromServer} from "../../components";
+import {tshirt} from '../../assets/images';
+
+const HomePage = () => {
+  const {state,dispatch} = useDataFromServer();
+
+  useEffect(async () => {
```
Comment (Debarshi95) [inline src/pages/home-page/HomePage.jsx]: Never make **useEffect callback async***.  To call async functions from useEffect, do it inside the normal callback function.

Read this for more info : https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret

Example :

``` 
   useEffect(()=>{ // Not an async function
      async function makeNetworkCall(){
       /// Make axios or fetch calls
       }
       makeNetworkCall(); // Call your async function here
    },[])
```

**Review by Debarshi95 — COMMENTED** [review]
Please work on the suggestions. Setup your vscode to use a formatter like Prettier and activate **formatOnSave** from Settings. This will auto format your code every time you hit CTRL+S. You can also add linter like eslint, styled guides to lint and format your code efficiently.

**Review by m26unkwn — COMMENTED** [review]
You should change your code as suggested by @Debarshi95 .

File: src/App.js (hunk 1 above)
Comment (shruthipatil11) [reply src/App.js:3]: okay. Spacing is provided.

File: src/assets/images/index.js (hunk 2 above)
Comment (shruthipatil11) [reply src/assets/images/index.js]: done.

**Review by Debarshi95 — APPROVED** [review]
Added a comment. Rest looks good. Work on the comment may be on next PR.

File: src/pages/home-page/HomePage.jsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
+import { useEffect } from "react";
+import axios from "axios";
+import "./home-page.css";
+import { Category, Header, useDataFromServer } from "../../components";
+import { tshirt } from "../../assets/images";
+
+const HomePage = () => {
+  const { state, dispatch } = useDataFromServer();
+
+  const fetchData = async () => {
```
Comment (Debarshi95) [inline src/pages/home-page/HomePage.jsx]: A suggestion, you can move this entire function block inside the useEffect callback just before calling the function.

**Review by Debarshi95 — APPROVED** [review]

File: src/pages/home-page/HomePage.jsx (hunk 3 above)
Comment (shruthipatil11) [reply src/pages/home-page/HomePage.jsx]: Thanks for feedback. Implemented it.

File: src/pages/home-page/HomePage.jsx (hunk 4 above)
Comment (shruthipatil11) [reply src/pages/home-page/HomePage.jsx]: I have made it as immediately invoked function.

File: src/pages/home-page/HomePage.jsx [hunk 5]
```diff
@@ -7,17 +7,15 @@ import { tshirt } from "../../assets/images";
...
-    fetchData();
+    (async () => {
+      try {
+        const response = await axios.get("/api/categories");
+        dispatch({ type: "saveData", payload: response.data.categories });
+      } catch (error) {
+        console.error(error, "couldn't fetch the featured categories");
+      }
+    })();
   }, []);
```
Comment (m26unkwn) [inline src/pages/home-page/HomePage.jsx]: This is good way to write it.

**Review by m26unkwn — APPROVED** [review]

**Pull request merged**